data/ignition/files/usr/local/bin/start-agent.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+source common.sh
+
+echo "Waiting for infra-env-id to be available"
+INFRA_ENV_ID=""
+until [[ $INFRA_ENV_ID != "" && $INFRA_ENV_ID != "null" ]]; do
+    sleep 5
+    printf 'Querying assisted-service for infra-env-id...'
+    INFRA_ENV_ID=$(curl {{.ServiceBaseURL}}/api/assisted-install/v2/infra-envs | jq .[0].id)
+done
+# trim quotes
+INFRA_ENV_ID=${INFRA_ENV_ID//\"}
+echo "Fetched infra-env-id and found: $INFRA_ENV_ID"
+
+# use infra-env-id to have agent register this host with assisted-service
+/usr/local/bin/agent --url {{.ServiceBaseURL}} --infra-env-id $INFRA_ENV_ID --agent-version quay.io/edge-infrastructure/assisted-installer-agent:latest --insecure=true
```

Comment:
At this point, there's nothing more to do in this shell script. Thus, there's little reason to keep it around, better exec into the agent instead of forking and running it. Also, quoting  for linter (We'll want to have a prow job that checks it.
```suggestion
exec /usr/local/bin/agent --url '{{.ServiceBaseURL}}' --infra-env-id "$INFRA_ENV_ID" --agent-version quay.io/edge-infrastructure/assisted-installer-agent:latest --insecure=true
```